src/main/java/com/sap/cloudfoundry/client/facade/adapters/RawCloudRoute.java

```diff
@@ -1,30 +1,56 @@
 package com.sap.cloudfoundry.client.facade.adapters;
 
+import org.cloudfoundry.client.v3.routes.Protocol;
 import org.cloudfoundry.client.v3.routes.Route;
 import org.immutables.value.Value;
 
-import com.sap.cloudfoundry.client.facade.domain.CloudDomain;
 import com.sap.cloudfoundry.client.facade.domain.CloudRoute;
-import com.sap.cloudfoundry.client.facade.domain.Derivable;
+import com.sap.cloudfoundry.client.facade.domain.ImmutableCloudDomain;
 import com.sap.cloudfoundry.client.facade.domain.ImmutableCloudRoute;
+import com.sap.cloudfoundry.client.facade.domain.ImmutableCloudMetadata;
+
+import java.util.UUID;
 
 @Value.Immutable
 public abstract class RawCloudRoute extends RawCloudEntity<CloudRoute> {
 
+    @Value.Parameter
     public abstract Route getRoute();
 
-    public abstract Derivable<CloudDomain> getDomain();
-
     @Override
     public CloudRoute derive() {
         Route route = getRoute();
+        String domainGuid = route.getRelationships()
+                                 .getDomain()
+                                 .getData()
+                                 .getId();
+        String domain = route.getUrl();
+        if (!route.getHost()
+                  .isEmpty()) {
+            domain = domain.substring(route.getHost()
+                                           .length() + 1);
+        }
+        if (!route.getPath()
+                  .isEmpty()) {
+            domain = domain.substring(0, domain.indexOf('/'));
+        }
+        if (route.getProtocol() == Protocol.TCP && route.getPort() != null) {
+            domain = domain.substring(0, domain.indexOf(':'));
```

Comment:
Consider this case:
  "protocol": "tcp",
  "port": 6666,
  "created_at": "2019-05-10T17:17:48Z",
  "updated_at": "2019-05-10T17:17:48Z",
  "host": "a-hostname",
  "path": "/some_path",
  "url": "a-hostname.a-domain.com/some_path",

The condition will be true, but the index of will return "-1"

Reply:
This is from CF API reference

Reply:
This should be an invalid route object. 
It should be `url: a-hostname.a-domain.com:6666/some_path`